src/solvers/unsteady_solver.C

```diff
@@ -232,6 +238,48 @@ void UnsteadySolver::retrieve_timestep()
      _system.get_dof_map().get_send_list());
 }
 
+void UnsteadySolver::integrate_adjoint_sensitivity(const QoISet & qois, const ParameterVector & parameter_vector, SensitivityData & sensitivities)
+{
+  // We are using the midpoint rule to integrate each timestep
+  // (f(t_j) + f(t_j+1))/2 (t_j+1 - t_j)
+
+  // Get t_j
+  Real time_left = _system.time;
+
+  // Left side sensitivities to hold f(t_j)
+  SensitivityData sensitivities_left(qois, _system, parameter_vector);
+
+  // Get f(t_j)
+  _system.adjoint_qoi_parameter_sensitivity(qois, parameter_vector, sensitivities_left);
+
+  // Advance to t_j+1
+  _system.time = _system.time + _system.deltat;
+
+  // Get t_j+1
+  Real time_right = _system.time;
+
+  // Right side sensitivities f(t_j+1)
+  SensitivityData sensitivities_right(qois, _system, parameter_vector);
+
+  // Remove the sensitivity rhs vector from system since we did not write it to file and it cannot be retrieved
+  _system.remove_vector("sensitivity_rhs0");
```

Comment:
Same issue as with the remove_vector elsewhere

Reply:
We'll have to live with this for now, these calls will be removed once the equationsystems::read changes are made.